tests/integration/api/mutations/checkout/anonymousCheckout.test.js

```diff
@@ -0,0 +1,501 @@
+const internalVariantIds = ["875", "874", "925"];
+
+const shopName = "Test Shop";
+
+const mockProduct = {
+  _id: internalProductId,
+  ancestors: [],
+  title: "Fake Product",
+  shopId: internalShopId,
+  isDeleted: false,
+  isVisible: true,
+  supportedFulfillmentTypes: ["shipping"]
+};
+
+const mockVariant = {
+  _id: internalVariantIds[0],
+  ancestors: [internalProductId],
+  attributeLabel: "Variant",
+  title: "Fake Product Variant",
+  shopId: internalShopId,
+  isDeleted: false,
+  isVisible: true
+};
+
+const mockOptionOne = {
+  _id: internalVariantIds[1],
+  ancestors: [internalProductId, internalVariantIds[0]],
+  attributeLabel: "Option",
+  title: "Fake Product Option One",
+  shopId: internalShopId,
+  isDeleted: false,
+  isVisible: true,
+  price: 19.99
+};
+
+const mockOptionTwo = {
+  _id: internalVariantIds[2],
+  ancestors: [internalProductId, internalVariantIds[0]],
+  attributeLabel: "Option",
+  title: "Fake Product Option Two",
+  shopId: internalShopId,
+  isDeleted: false,
+  isVisible: true,
+  price: 29.99
+};
+
+const mockShippingMethod = {
+  _id: "mockShippingMethod",
+  name: "Default Shipping Provider",
+  shopId: internalShopId,
+  provider: {
+    enabled: true,
+    label: "Flat Rate",
+    name: "flatRates"
+  },
+  methods: [
+    {
+      cost: 2.5,
+      fulfillmentTypes: [
+        "shipping"
+      ],
+      group: "Ground",
+      handling: 1.5,
+      label: "Standard mockMethod",
+      name: "mockMethod",
+      rate: 1,
+      _id: "mockMethod",
+      enabled: true
+    }
+  ]
+};
+
+
+let testApp;
+let addCartItems;
+let anonymousCartByCartQuery;
+let availablePaymentMethods;
+let createCart;
+let removeCartItems;
+let placeOrder;
+let publishProducts;
+let selectFulfillmentOptionForGroup;
+let setEmailOnAnonymousCart;
+let setShippingAddressOnCart;
+let updateCartItemsQuantity;
+let updateFulfillmentOptionsForGroup;
+
+beforeAll(async () => {
+  testApp = new TestApp();
+  await testApp.start();
+
+  addCartItems = testApp.mutate(AddCartItemsMutation);
+  anonymousCartByCartQuery = testApp.mutate(AnonymousCartByCartIdQuery);
+  availablePaymentMethods = testApp.query(AvailablePaymentMethodsQuery);
+  createCart = testApp.mutate(CreateCartMutation);
+  removeCartItems = testApp.mutate(RemoveCartItemsMutation);
+  placeOrder = testApp.mutate(PlaceOrderMutation);
+  publishProducts = testApp.mutate(PublishProductToCatalogMutation);
+  removeCartItems = testApp.mutate(RemoveCartItemsMutation);
+  setEmailOnAnonymousCart = testApp.mutate(SetEmailOnAnonymousCart);
+  selectFulfillmentOptionForGroup = testApp.mutate(SelectFulfillmentOptionForGroupMutation);
+  setShippingAddressOnCart = testApp.mutate(SetShippingAddressOnCartMutation);
+  updateCartItemsQuantity = testApp.mutate(UpdateCartItemsQuantityMutation);
+  updateFulfillmentOptionsForGroup = testApp.mutate(UpdateFulfillmentOptionsForGroupMutation);
+
+  await testApp.insertPrimaryShop({
```

Comment:
seeing this error when running the tests
```
 Error in "afterOrderCreate" consumer Error: No email template orders/new found for language mockLanguage
```
this is due to the fact that the email templates do not exist in the database. They are created when the `createShop` mutation is executed. For tests such as this one that need email templates created, use the `createShop` mutation instead of in inserting the shop directly into the db. Further, to set other settings, use `updateOne` after the shop is created. This issue is present in all three tests.


only admins can create shops, so you will need a mock admin account 
```
const mockAdminAccount = Factory.Account.makeOne({
  _id: "mockAdminAccountId",
  roles: {
    [internalShopId]: ["admin"]
  },
  shopId: internalShopId
});
```

```
  await testApp.createUserAndAccount(mockAdminAccount);
  await testApp.setLoggedInUser(mockAdminAccount);
 await testApp.context.mutations.createShop({ ...testApp.context, isInternalCall: true }, {
    name: shopName,
    shopId: internalShopId
  });

  // Set other settings
  await testApp.collections.Shops.updateOne(
    { _id: internalShopId },
    { $set: { availablePaymentMethods: ["iou_example"] } },
    { $set: { allowGuestCheckout: true } }
  );
```

for a complete reference checkout this test: https://github.com/reactioncommerce/reaction/blob/release-3.0.0/tests/integration/api/mutations/inviteShopMember/inviteShopMember.test.js#L49